spec/lib/migrations/load_rate_reference_spec.rb

```diff
@@ -2,51 +2,74 @@
 require 'rake'
 require 'roo'
 
+RSpec.shared_examples "a rate reference" do |attributes|
+  attributes.each do |attribute, value|
+    it "should return #{value} from ##{attribute}" do
+      expect(subject.send(attribute)).to eq(value)
+    end
+  end
+end
+
 RSpec.describe 'Load Rating Regions Task', :type => :task do
 
   context "rate_reference:update_rating_regions" do
     let(:file_path) { File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx") }
-    let(:subject) { Roo::Spreadsheet.open(file_path) }
 
     before :all do
       Rake.application.rake_require "tasks/migrations/load_rate_reference"
       Rake::Task.define_task(:environment)
     end
-    it 'expect not raise errors' do
-      expect { invoke_task }.not_to raise_error
-    end
 
-    it 'expect to read xlsx file' do
-      expect(subject.sheet(0).row(2)).to match_array(["01001", "Hampden", "No", "Rating Area 1"])
-      expect(subject.sheet(0).row(3)).to match_array(["01002", "Hampshire", "Yes", "Rating Area 1"])
-      expect(subject.sheet(0).row(4)).to match_array(["01002", "Franklin", "Yes", "Rating Area 1"])
-      expect(subject.sheet(0).row(699)).to match_array(["02791", "Bristol", "No", "Rating Area 6"])
+    before :context do
+      invoke_task
     end
 
-    it "should have the right data" do
-      expect(subject.sheet(0).row(2).first.to_i).to eq RateReference.first.zip_code
-      expect(subject.sheet(0).row(2)[1]).to eq RateReference.first.county
-      expect(subject.sheet(0).row(2).last).to eq RateReference.first.rating_region
+    context "it creates RateReference elements correctly" do
+      subject { RateReference.first }
+      it_should_behave_like "a rate reference", { zip_code: "01001",
```

Comment:
This context is clear - we're testing that it creates an element correctly and we define the subject explicitly.